packages/dashboards/addon/components/dashboard-action-list.js

```diff
@@ -1,14 +1,42 @@
 /**
- * Copyright 2017, Yahoo Holdings Inc.
+ * Copyright 2018, Yahoo Holdings Inc.
  * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
  */
-import Ember from 'ember';
+import Component from '@ember/component';
 import layout from '../templates/components/dashboard-action-list';
+import { get } from '@ember/object';
+import { inject as service } from '@ember/service';
 
-export default Ember.Component.extend({
+export default Component.extend({
   layout,
 
+  /**
+   * @property {String} tagName
+   */
   tagName: 'ul',
 
-  classNames: [ 'actions' ]
+  /**
+   * @property {Array} classNames
+   */
+  classNames: [ 'actions' ],
+
+  /**
+   * @property {Service} router
+   */
+  router: service(),
+
+  actions: {
+    /**
+     * @action buildDashboardUrl
+     * @param {Object} dashboard - model with id
+     * @returns {String} url for given dashboard
+     */
+    buildDashboardUrl(dashboard) {
```

Comment:
Isn't this also similar to the method for reports?
Is it possible to extend that and make use of it here, since our route structures are very similar.

Reply:
I suppose I could create a base component that both `navi-action-list` and `dashboard-action-list` extend from, but that seems unnecessary two combine two small actions that were separately defined in the routes before my changes.

Reply:
You don't need a base component, how about something simple like a mixin with an action/method that would have the buildUrl logic.